src/core/thread/key_manager.hpp

```diff
@@ -504,6 +574,11 @@ class KeyManager : public InstanceLocator, private NonCopyable
     void        StartKeyRotationTimer(void);
```

Comment:
In the code above we have
```c++
    OT_TOOL_PACKED_BEGIN
    struct Keys
    {
        Mle::Key mMleKey;
        Mac::Key mMacKey;
    } OT_TOOL_PACKED_END;

    union HashKeys
    {
        Crypto::HmacSha256::Hash mHash;
        Keys                     mKeys;
    };
```

How about changing the `Keys` type to use `otMacKey` (which is the literal key) instead of the union types `Mac:Key`? (it may also help to actually define a `Mac::LiteralKey` type as mirror of `otMacKey` (with `Clear()` and other helper methods) which we use here).

Reply:
We can't do this, as we pass `Keys.mMacKey` to sub_mac as Previous, Current and Next keys. And these need to be unions.